bindings_ffi/src/mls.rs

```diff
@@ -114,6 +119,73 @@ pub async fn create_client(
     }))
 }
 
+#[derive(uniffi::Object)]
+pub struct FfiSignatureRequest {
+    // Using `tokio::sync::Mutex`bc MutexGuard cannot be sent between threads.
+    inner: Arc<tokio::sync::Mutex<SignatureRequest>>,
+}
+
+#[uniffi::export(async_runtime = "tokio")]
+impl FfiSignatureRequest {
+    // Signature that's signed by EOA wallet
+    pub async fn add_ecdsa_signature(&self, signature_bytes: Vec<u8>) -> Result<(), GenericError> {
+        let mut inner = self.inner.lock().await;
+        let signature_text = inner.signature_text();
+        inner
+            .add_signature(Box::new(RecoverableEcdsaSignature::new(
+                signature_text,
+                signature_bytes,
+            )))
+            .await?;
+
+        Ok(())
+    }
+
+    pub async fn add_erc1271_signature(
+        &self,
+        signature_bytes: Vec<u8>,
+        address: String,
+        chain_rpc_url: String,
+    ) -> Result<(), GenericError> {
+        let mut inner = self.inner.lock().await;
+        let erc1271_siganture = Erc1271Signature::new_with_rpc(
+            inner.signature_text(),
+            signature_bytes,
+            address,
+            chain_rpc_url,
+        )
+        .await?;
+        inner.add_signature(Box::new(erc1271_siganture)).await?;
+        Ok(())
+    }
+
+    pub async fn signature_text(&self) -> Result<String, GenericError> {
+        Ok(self.inner.lock().await.signature_text())
+    }
+
+    /// missing signatures that are from [MemberKind::Address]
+    pub async fn missing_address_signatures(
```

Comment:
You'd probably be better off just converting these addresses to strings rather than returning a `FfiMemberIdentifier` type.

Right now `FfiMemberIdentifier` doesn't even do anything, so there's no way to get the address out of it. But I think the only thing the caller _would_ want to do is to get the address as a string.

Reply:
No, I just didn't think of converting them to strings. Will do what you suggested